app/components/button.js

```diff
@@ -3,7 +3,17 @@
 import React from 'react'
 import styled from 'styled-components'
 
-const BaseButton = styled.button`
+const BaseButton = styled.button.attrs({
+  onClick: props => {
+    if (props.trigger) {
+      return function (e) {
+        e.stopPropagation()
+        props.trigger.apply(this, [e])
+      }
+    }
+    return undefined
+  }
```

Comment:
I'd suggest a minor change (reverse condition + use call to avoid declaring an array):

```
props => {
  if (!props.trigger) return
  return function (e) {
      e.stopPropagation()
      props.trigger.call(this, e)
  }
}
```